### PR TITLE
Added BREAK_TIME attendance functionality in punchIn/Out

### PR DESCRIPTION
**Pull request by rokfirst**
Extended AttendanceRecord and PunchIn/Out services to support BREAK_TIME attendances

### CODE REVIEW

File: installer/config/Conf.tpl.php [hunk 1]
```diff
@@ -28,12 +28,15 @@ class Conf
...
+
         $this->dbPort = '{{dbPort}}';
         if (defined('ENVIRONMENT') && ENVIRONMENT == 'test') {
             $prefix = defined('TEST_DB_PREFIX') ? TEST_DB_PREFIX : '';
             $this->dbName = $prefix . 'test_{{dbName}}';
         } else {
-            $this->dbName = '{{dbName}}';
+            $this->dbName = 'orangehrm'; 
+            // $this->dbName = '{{dbName}}'
         }
```
Comment (vrenko007) [inline installer/config/Conf.tpl.php:40]: I dislike hardcoding prod dbName here, maybe do the next suggestion for production environment

```suggestion
        } else if (defined('ENVIRONMENT') && ENVIRONMENT == 'production') {
            $this->dbName = '{{dbName}}';
        } else {
            $this->dbName = 'orangehrm';
        }
```

File: src/client/src/orangehrmAttendancePlugin/components/RecordAttendance.vue [hunk 2]
```diff
@@ -308,17 +405,16 @@ export default {
...
           this.attendanceRecord.timezone?._offset ?? timezone.offset,
         timezoneName: this.attendanceRecord.timezone?.id ?? timezone.name,
+        breakNote: this.attendanceRecord.breakNote,
       };
       try {
-        if (
-          this.latestAttendanceRecord &&
-          this.latestAttendanceRecord?.attendanceType.id === 'WORK_TIME'
-        ) {
+        /*if (this.latestAttendanceRecord?.attendanceType?.id === 'WORK_TIME') {
```
Comment (vrenko007) [inline src/client/src/orangehrmAttendancePlugin/components/RecordAttendance.vue:417]: Could we remove the documented code here, as it unnecessarily complicates the function and is not used?

File: src/plugins/orangehrmAdminPlugin/config/supported_languages.yml [hunk 3]
```diff
@@ -10,6 +10,7 @@ languages:
   6: {key: 'de', value: 'German - Deutsch'}
   7: {key: 'es', value: 'Spanish - Español'}
   8: {key: 'es_CR', value: 'Spanish (Costa Rica) - Español (Costa Rica)'}
+  9: {key: 'SI', value: 'Slovenian'}
```
Comment (vrenko007) [inline src/plugins/orangehrmAdminPlugin/config/supported_languages.yml:13]: Is capitalized key here correct or should it be `si`?

File: src/plugins/orangehrmAttendancePlugin/Api/EmployeeLatestAttendanceRecordAPI.php [hunk 4]
```diff
@@ -73,12 +76,36 @@ public function getOne(): EndpointResult
             CommonParams::PARAMETER_EMP_NUMBER,
             $this->getAuthUser()->getEmpNumber()
         );
+
+
+
```
Comment (vrenko007) [inline src/plugins/orangehrmAttendancePlugin/Api/EmployeeLatestAttendanceRecordAPI.php:81]: Is there any reason here for three empty lines?

File: src/plugins/orangehrmAttendancePlugin/Dao/AttendanceDao.php [hunk 5]
```diff
@@ -363,20 +391,23 @@ private function getAttendanceReportQueryBuilderWrapper(
             // from date is not null and to date is null
             $q->leftJoin('employee.attendanceRecords', 'attendanceRecord', Expr\Join::WITH, $q->expr()->andX(
                 $q->expr()->gte('attendanceRecord.punchInUserTime', ':fromDate')
-            ));
+            )
+            );
```
Comment (vrenko007) [inline src/plugins/orangehrmAttendancePlugin/Dao/AttendanceDao.php:395]: I am not sure what is convention here in PHP, but it looks janky for me to be like this, could we either revert this change or change the call to the topmost function to be multiline

File: src/plugins/orangehrmAttendancePlugin/Dao/AttendanceDao.php [hunk 6]
```diff
@@ -363,20 +391,23 @@ private function getAttendanceReportQueryBuilderWrapper(
...
             $q->setParameter('fromDate', $attendanceReportSearchFilterParams->getFromDate());
         } elseif (is_null($attendanceReportSearchFilterParams->getFromDate()) && !is_null($attendanceReportSearchFilterParams->getToDate())) {
             // from date is null and to date is not null
             $q->leftJoin('employee.attendanceRecords', 'attendanceRecord', Expr\Join::WITH, $q->expr()->andX(
                 $q->expr()->lte('attendanceRecord.punchOutUserTime', ':toDate')
-            ));
+            )
+            );
```
Comment (vrenko007) [inline src/plugins/orangehrmAttendancePlugin/Dao/AttendanceDao.php:402]: See previous comment

File: src/plugins/orangehrmAttendancePlugin/Dao/AttendanceDao.php [hunk 7]
```diff
@@ -363,20 +391,23 @@ private function getAttendanceReportQueryBuilderWrapper(
...
             $q->setParameter('toDate', $attendanceReportSearchFilterParams->getToDate());
         } elseif (!is_null($attendanceReportSearchFilterParams->getFromDate()) && !is_null($attendanceReportSearchFilterParams->getToDate())) {
             // both from date and to date is not null
             $q->leftJoin('employee.attendanceRecords', 'attendanceRecord', Expr\Join::WITH, $q->expr()->andX(
                 $q->expr()->gte('attendanceRecord.punchInUserTime', ':fromDate'),
                 $q->expr()->lte('attendanceRecord.punchOutUserTime', ':toDate')
-            ));
+            )
+            );
```
Comment (vrenko007) [inline src/plugins/orangehrmAttendancePlugin/Dao/AttendanceDao.php:410]: See previous comment

File: src/plugins/orangehrmAttendancePlugin/Dao/AttendanceDao.php [hunk 8]
```diff
@@ -602,7 +633,8 @@ private function getEmployeeAttendanceSummaryQueryBuilderWrapper(EmployeeAttenda
         $q->leftJoin('employee.attendanceRecords', 'attendanceRecord', Expr\Join::WITH, $q->expr()->andX(
             $q->expr()->gte('attendanceRecord.punchInUserTime', ':fromDate'),
             $q->expr()->lte('attendanceRecord.punchInUserTime', ':toDate')
-        ));
+        )
+        );
```
Comment (vrenko007) [inline src/plugins/orangehrmAttendancePlugin/Dao/AttendanceDao.php:637]: See previous comment

**Review by vrenko007 — CHANGES_REQUESTED** [review]

File: src/plugins/orangehrmAdminPlugin/config/supported_languages.yml (hunk 3 above)
Comment (rokfirst) [reply src/plugins/orangehrmAdminPlugin/config/supported_languages.yml:13]: tested with SI, it worked, tested with si, didn't work

File: src/plugins/orangehrmAttendancePlugin/Dao/AttendanceDao.php (hunk 5 above)
Comment (rokfirst) [reply src/plugins/orangehrmAttendancePlugin/Dao/AttendanceDao.php:395]: this is my subconsciousness spamming alt + f 😄 . will revert

**Review by vrenko007 — APPROVED** [review]
